apps/api/src/domain/models/song/commands/add-lyrics-for-song/add-lyrics-for-song.command.integration.spec.ts

```diff
@@ -0,0 +1,94 @@
+import { NotFound } from '../../../../../lib/types/not-found';
+import TestRepositoryProvider from '../../../../../persistence/repositories/__tests__/TestRepositoryProvider';
+import generateDatabaseNameForTestSuite from '../../../../../persistence/repositories/__tests__/generateDatabaseNameForTestSuite';
+import { DTO } from '../../../../../types/DTO';
+import { assertCommandSuccess } from '../../../__tests__/command-helpers/assert-command-success';
+import { CommandAssertionDependencies } from '../../../__tests__/command-helpers/types/CommandAssertionDependencies';
+import { dummySystemUserId } from '../../../__tests__/utilities/dummySystemUserId';
+
+const commandType = 'ADD_LYRICS_FOR_SONG';
+
+const existingSong = getValidAggregateInstanceForTest(AggregateType.song).clone({
+    lyrics: undefined,
+});
+
+const validPayload: AddLyricsForSong = {
+    aggregateCompositeIdentifier: existingSong.getCompositeIdentifier(),
+    lyrics: existingSong.lyrics,
+    languageCode: LanguageCode,
+};
+
+const validCommandFSA = {
+    type: commandType,
+    payload: validPayload,
+};
+
+const buildValidCommandFSA = (): FluxStandardAction<DTO<AddLyricsForSong>> => validCommandFSA;
+
+describe(commandType, () => {
+    let app: INestApplication;
+
+    let testRepositoryProvider: TestRepositoryProvider;
+
+    let commandHandlerService: CommandHandlerService;
+
+    let idManager: IIdManager;
+
+    let commandAssertionDependencies: CommandAssertionDependencies;
+
+    beforeAll(async () => {
+        ({ testRepositoryProvider, commandHandlerService, idManager, app } =
+            await setUpIntegrationTest({
+                ARANGO_DB_NAME: generateDatabaseNameForTestSuite(),
+            }).catch((error) => {
+                throw error;
+            }));
+
+        commandAssertionDependencies = {
+            testRepositoryProvider,
+            idManager,
+            commandHandlerService,
+        };
+    });
+
+    afterAll(async () => {
+        await app.close();
+    });
+
+    beforeAll(async () => {
+        await testRepositoryProvider.testSetup();
+    });
+
+    describe('When the command is valid', () => {
+        it('should succeed', async () => {
+            await assertCommandSuccess(commandAssertionDependencies, {
+                systemUserId: dummySystemUserId,
+                buildValidCommandFSA,
+                initialState: new DeluxeInMemoryStore({
+                    [AggregateType.song]: [existingSong],
+                }).fetchFullSnapshotInLegacyFormat(),
+                checkStateOnSuccess: async ({
+                    aggregateCompositeIdentifier: { id: SongCompositeId },
+                }: AddLyricsForSong) => {
+                    const songSearchResult = await testRepositoryProvider
+                        .forResource(ResourceType.song)
+                        .fetchById(SongCompositeId);
+
+                    expect(songSearchResult).not.toBe(NotFound);
```

Comment:
Later on, you can also check that the event record is persisted properly. There's a helper for that if you look at some of the other create commands (maybe some are missing this too).